### ISSUE TITLE
[Serve] Controller gets stuck and makes no progress when deployment has invalid placement group bundle 

### ISSUE THREAD

**Issue by shrekris-anyscale** [ray-project/ray#43888]
### What happened + What you expected to happen

If a deployment has an invalid placement group bundle, the controller gets stuck in an infinite loop and cannot make progress reconciling any deployments. The only workaround is killing the controller with `kill -9 [controller PID]`. See reproduction script for details.

### Versions / Dependencies

Ray on the latest `master`.

### Reproduction script

This app contains an invalid placement group bundle.

```python
# File name: repro.py

from ray import serve

# {"CPU": 0, "GPU": 0} is an invalid placement_group_bundle
@serve.deployment(
    placement_group_bundles=[{"CPU": 0, "GPU": 0}, {"GPU": 1.0, "accelerator_type_a10": 0.01, "CPU": 8.0}],
    placement_group_strategy="STRICT_PACK",
    ray_actor_options={"num_cpus": 0},
)
def f():
    return "hi"

app = f.bind()
```

Start a long-lived Ray cluster:

```
ray start --head
```

Run the app:

```
serve run repro:app
```

The app will fail to launch with the following error printed repeatedly:

```
(ServeController pid=86229) ERROR 2024-03-11 14:20:12,302 controller 86229 controller.py:404 - Exception updating deployment state.
(ServeController pid=86229) Traceback (most recent call last):
(ServeController pid=86229)   File "/Users/shrekris/Desktop/ray/python/ray/serve/_private/controller.py", line 390, in run_control_loop
(ServeController pid=86229)     any_recovering = self.deployment_state_manager.update()
(ServeController pid=86229)   File "/Users/shrekris/Desktop/ray/python/ray/serve/_private/deployment_state.py", line 2761, in update
(ServeController pid=86229)     deployment_to_replicas_to_stop = self._deployment_scheduler.schedule(
(ServeController pid=86229)   File "/Users/shrekris/Desktop/ray/python/ray/serve/_private/deployment_scheduler.py", line 640, in schedule
(ServeController pid=86229)     self._schedule_replica(
(ServeController pid=86229)   File "/Users/shrekris/Desktop/ray/python/ray/serve/_private/deployment_scheduler.py", line 534, in _schedule_replica
(ServeController pid=86229)     pg = self._create_placement_group_fn(
(ServeController pid=86229)   File "/Users/shrekris/Desktop/ray/python/ray/_private/client_mode_hook.py", line 141, in wrapper
(ServeController pid=86229)     return func(*args, **kwargs)
(ServeController pid=86229)   File "/Users/shrekris/Desktop/ray/python/ray/util/placement_group.py", line 230, in placement_group
(ServeController pid=86229)     raise ValueError(
(ServeController pid=86229) ValueError: Bundles cannot be an empty dictionary or resources with only 0 values. Bundles: [{'CPU': 0.0, 'GPU': 0.0}, {'GPU': 1.0, 'accelerator_type_a10': 0.01, 'CPU': 8.0}]
```

Try running `serve shutdown` in another terminal. The controller will receive the shutdown request but will not shut down.

```
serve shutdown -y -a http://localhost:8265
```

Since we started the app with `serve run`, spamming ctrl-C will kill the app. If we started the app using a config file instead, we would have to kill the controller with `kill -9 [controller PID]`.

### Issue Severity

High: It blocks me from completing my task.

**Comment by zcin** [comment]
@jjyao Seems like if the placement group bundles are invalid, we try to create them with `ray.util.placement_group(bundles)` and it fails, then the controller will just busy spin while failing repeatedly while calling `deployment_scheduler.schedule()`.

**Comment by jjyao** [comment]
Yea, we should catch the error. Ideally we should validate pg in the @serve.deployment so we can throw errors early.

**Comment by anyscalesam** [comment]
@jjyao have https://github.com/anyscale/runtime/issues/472 to discuss next sprint planning on Mon.

**Comment by shrekris-anyscale** [comment]
Closed by #44026.
